### PR TITLE
Fixed build problems with GCC 6

### PR DESCRIPTION
**Pull request by jasp00**
Most problems are about code formatting. The problem in `modules_limit.cpp` is about type promotion.

### CODE REVIEW

**Comment by tresf** [comment]
@Umcaruje, now these Zyn changes would technically qualify as upstream code changes.  I'm not sure how widespread Zyn 2.4 codebase is these days (probably not worth committing upstream due to its age).  If someone disagrees, we can ping `@fundamental`.

```diff
+ plugins/zynaddsubfx/zynaddsubfx/src/UI/EnvelopeUI.fl
+ plugins/zynaddsubfx/zynaddsubfx/src/UI/ResonanceUI.fl
```

File: plugins/opl2/fmopl.c [hunk 1]
```diff
@@ -653,29 +653,28 @@ static void init_timetables( FM_OPL *OPL , int ARRATE , int DRRATE ) {
 	double rate;
 
 	/* make attack rate & decay rate tables */
-	for ( i = 0; i < 4; i++ ) OPL->AR_TABLE[i] = OPL->DR_TABLE[i] = 0; {
```
Comment (tresf) [inline plugins/opl2/fmopl.c]: Had to read this about 5 times to figure out what was happening in the original.  Thanks.

**Comment by tresf** [comment]
Looks good, thanks!  All changes are benign.  The `for loop` threw me for a... well loop. 😄

**Comment by midi-pascal** [comment]
@jasp00 @tresf The commit 60740fb breaks the build on my Ubuntu 14.04 x64 with this error:
``
[ 78%] Generating moc_ZynAddSubFx.cxx
Generating embedded_resources.h
Linking CXX executable ../RemoteZynAddSubFx
Scanning dependencies of target zynaddsubfx
[100%] Built target lmmsobjs
[100%] [100%] Building CXX object plugins/zynaddsubfx/CMakeFiles/zynaddsubfx.dir/ZynAddSubFx.cpp.o
Scanning dependencies of target lmms
Building CXX object plugins/zynaddsubfx/CMakeFiles/zynaddsubfx.dir/moc_ZynAddSubFx.cxx.o
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libfltk.a(Fl_Double_Window.o): undefined reference to symbol 'XdbeAllocateBackBufferName'
//usr/lib/x86_64-linux-gnu/libXext.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [plugins/RemoteZynAddSubFx] Error 1
``
make[1]: *** [plugins/zynaddsubfx/CMakeFiles/RemoteZynAddSubFx.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[100%] [100%] Building CXX object src/CMakeFiles/lmms.dir/lmms_automoc.cpp.o
Building CXX object src/CMakeFiles/lmms.dir/core/main.cpp.o
Linking CXX executable ../lmms
[100%] Built target lmms
Linking CXX shared module ../libzynaddsubfx.so
[100%] Built target zynaddsubfx
make: *** [all] Error 2

**Comment by tresf** [comment]
@midi-pascal 

1. Use code block tags please (\`\`\`)
2. Wrong PR :)

**Comment by midi-pascal** [comment]
@tresf I fixed the formating but, the PR number is the one you gave me in your email (and the one I reverted and make the build succeed)...

> Can you try to revert commits 60740fb, a06cb01?

**Comment by tresf** [comment]
Ah yes... critical mistake.  Fixed. 😈 

```diff
Can you try to revert commits 60740fb, a06cb01?
-If either fix it, please post in #2747 or #2749, respectively (or feel free to open a bug report on it).
+If either fix it, please post in #2749 or #2747, respectively (or feel free to open a bug report on it).
                                      ^--------^
```

**Comment by midi-pascal** [comment]
@tresf Oh I just see it is not the right issue... I will move this to 2749. Sorry.

**Pull request merged**